packages/now-routing-utils/src/superstatic.ts

```diff
@@ -169,6 +173,14 @@ function replaceSegments(segments: string[], destination: string): string {
     }
 
     for (const [name, value] of Object.entries(indexes)) {
+      const segmentRegex = new RegExp(`\\${value}(?!\\d)`);
+      if (isRedirect && segmentRegex.test(pathname + (hash || ''))) {
```

Comment:
```suggestion
      if (isRedirect && new RegExp(`\\${value}(?!\\d)`).test(pathname + (hash || ''))) {
```

The only change I would make here is avoid creating a new RegExp if its not a redirect.